app/src/views/IntakeProfile/constants/index.ts

```diff
@@ -0,0 +1,179 @@
+import {faker} from '@faker-js/faker';
+import {array, object, string} from 'yup';
+import {InitialValues} from '..';
+import {
+  FieldGroup,
+  Fields,
+  fieldTypes,
+  FieldTypes,
+  Response,
+} from '../../../services/profile';
+
+export const fieldGroupBuilder = (
+  options: Partial<FieldGroup> = {},
+): FieldGroup => ({
+  id: faker.string.numeric(4),
+  title: faker.lorem.sentence({min: 2, max: 4}),
+  fields: [],
+  ...options,
+});
+
+export const fieldBuilder = (options: Partial<Fields> = {}): Fields => ({
+  id: faker.string.numeric(4),
+  title: faker.lorem.sentence({min: 5, max: 10}),
+  type: faker.helpers.arrayElement(fieldTypes),
+  ...options,
+  properties: {
+    ...options.properties,
+  },
+  validations: {
+    ...options.validations,
+  },
+});
+
+/**
+ * Creates an object used for the initial Formik valiues
+ * It takes the form of:
+ * {
+ *  fieldGroupId: {
+ *     fieldId: responseValue
+ *  }
+ * }
+ */
+const fieldDefaultValue = (fieldType: FieldTypes) => {
+  switch (fieldType) {
+    case 'short_text':
+      return '';
+    case 'long_text':
+      return '';
+    case 'dropdown':
+      return '';
+    case 'number':
+      return '';
+    case 'additional_guests':
+      return [];
+    case 'email':
+      return '';
+    case 'multiple_choice':
+      return '';
+    case 'yes_no':
+      return '';
+    default:
+      return '';
+  }
+};
+
+export const createInitialValues = (
+  fieldGroups: FieldGroup[],
+  responses: Response[],
+): InitialValues => {
+  return fieldGroups.reduce((acc: InitialValues, fieldGroup) => {
+    const fields = fieldGroup.fields.reduce((acc, field) => {
+      return {
+        ...acc,
+        [field.id]:
+          responses.find(response => response.fieldId === field.id)?.value ||
+          fieldDefaultValue(field.type),
+      };
+    }, {});
+
+    return {
+      ...acc,
+      [fieldGroup.id]: {...fields},
+    };
+  }, {});
+};
+
+/**
+ * Creates a validation schema for Formik based on field type
+ * It takes the form of:
+ * {
+ *  fieldGroupId: {
+ *     fieldId: validationSchema
+ *  }
+ * }
+ */
+
+const phoneRegExp =
+  /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
+
+export const typeValidations = {
+  short_text: string(),
+  long_text: string(),
+  number: string().matches(phoneRegExp, 'phone number is not valid'),
+  email: string().email('Must be a valid email address'),
+  yes_no: string(),
+  dropdown: string(),
+  multiple_choice: string(),
+  additional_guests: array().of(
+    object().shape({
+      name: string().required('Name is required'),
+      dob: string().required('DOB is required'),
```

Comment:
One thing that we can add here is more redundant date validation since standardizing date formatting will help clean up our input data for the matching algo. 
We already have date-fns as a dependency and it would be helpful to include it here. I used it on ``app/src/utils/GuestApplicationSchema.ts`` line 12 for a quick reference

Reply:
Could also be its own "type" since it's used on the user themselves and in dropdowns like this one.

Reply:
Thanks @JpadillaCoding working on this now.

Reply:
Hey, @JpadillaCoding I made a ticket for this issue since it's taking a bit longer than I thought and I don't want hold this PR up for too much longer. Here is the ticket: https://github.com/hackforla/HomeUniteUs/issues/713